api/app/tests/hfi/test_hfi_endpoint_request.feature

```diff
@@ -10,27 +10,48 @@ Feature: /hfi/
+            | /api/hfi-calc/fire_centre/1/2020-05-21/2020-05-26                                 | post | 200         | hfi/test_hfi_endpoint_set_date_range_response.json   | True          | None                                  |
+            | /api/hfi-calc/fire_centre/1/2020-05-21/2020-05-26                                 | post | 200         | hfi/test_hfi_endpoint_set_date_range_response.json   | True          | test_hfi_endpoint_stored_request.json |
             # pdf
-            | api/hfi-calc/fire_centre/1/2020-05-21/pdf                                                        | get  | 200         | None                                                     | False         | None                                  |
-            | api/hfi-calc/fire_centre/1/2020-05-21/pdf                                                        | get  | 200         | None                                                     | False         | test_hfi_endpoint_stored_request.json |
+            | api/hfi-calc/fire_centre/1/2020-05-21/pdf                                         | get  | 200         | None                                                 | False         | None                                  |
+            | api/hfi-calc/fire_centre/1/2020-05-21/pdf                                         | get  | 200         | None                                                 | False         | test_hfi_endpoint_stored_request.json |
+
+    Scenario: HFI - request set fire starts with role
+        Given I have a stored request <stored_request_json>
+        And I spy on store_hfi_request
+        And I received a hfi-calc <url> with <verb>
+        Then the response status code is <status_code>
+        And the response is <response_json>
+        And the response isn't cached
+        And request == saved = <request_saved>
 
+        Examples:
+            | url                                                                                              | verb | status_code | response_json                                            | request_saved | stored_request_json                   |
+            | /api/hfi-calc/fire_centre/1/2020-05-21/planning_area/1/fire_starts/2020-05-21/fire_start_range/2 | post | 200         | hfi/test_hfi_endpoint_response_set_fire_start_range.json | True          | None                                  |
+            | /api/hfi-calc/fire_centre/1/2020-05-21/planning_area/1/fire_starts/2020-05-21/fire_start_range/2 | post | 200         | hfi/test_hfi_endpoint_response_set_fire_start_range.json | True          | test_hfi_endpoint_stored_request.json |
 
+    Scenario: HFI - request set fire starts without role
+        Given I have a stored request <stored_request_json>
+        And I spy on store_hfi_request
```

Comment:
I think you'd be better off having a `Given request with roles: <roles>` statement. That way you don't need multiple scenarios, and you can specify the roles you want as input.
If my comment doesn't make sense, please reach out.

Reply:
It's difficult to do with the specific JWT mock I need, not sure if there's something easier I'm missing

Reply:
Oh yeah - you can get that  mock in place without doing a decorator on a function - I can explain better with a screen share - reach out.